### PR TITLE
keycloak: add missing `validate_certs` parameters for `open_url` calls

### LINKED ISSUE #4391: Keycloak authentication - validate_certs value is ignored
### Summary

community.general.keycloak_authentication module ignore value `False` and I can't disable TLS validation for self signed instance without adding certificate into trust store. I think this should works by default as this is useful for development instances with self signed cert. 

### Issue Type

Bug Report

### Component Name

keycloak_authentication

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.9.27
  config file = /home/myuser/.ansible.cfg
  configured module search path = ['/home/myuser/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.10.2 (main, Jan 17 2022, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

Fedora release 35 (Thirty Five)

### Steps to Reproduce

```yaml (paste below)
- name: Modify authentication flow for browser
  community.general.keycloak_authentication:
    auth_keycloak_url: "{{ keycloak_configuration_auth_url }}"
    auth_realm: "{{ keycloak_configuration_auth_realm }}"
    auth_username: "{{ keycloak_configuration_auth_username }}"
    auth_password: "{{ keycloak_configuration_auth_password }}"
    validate_certs: False
    realm: "{{ auth.realm }}"
    alias: "{{ auth.alias }}"
    copyFrom: "{{ auth.copyFrom }}"
    description: "{{ auth.description | default(omit) }}"
    authenticationExecutions:
      "{{ auth.authenticationExecutions | default(omit) }}"
    force: "{{ auth.force | default(omit) }}"
    providerId: "{{ auth.providerId | default(omit) }}"
    state: present
  register: auth_results
  delegate_to: localhost
  loop: "{{ keycloak_configuration_authentication }}"
  loop_control:
    loop_var: auth
```

### Expected Results

Task play should use `False` value and ignore not trusted certificate.

### Actual Results

```console (paste below)
` <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for`
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by lmeunier**
##### SUMMARY

As stated in the [documentation](https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_authentication_module.html#parameter-validate_certs), the `validate_certs` parameter can be
used to verify (or not) the TLS certificates. But, for some modules (at
least for the `keycloak_authentication` module), this parameter is not
used with the `open_url` function.

Fixes #4391.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

keycloak_authentication

##### ADDITIONAL INFORMATION

before:
```paste below
TASK [keycloak : create custom authentication flow] *******************************************************************************************************************************************************************************************
fatal: [********]: FAILED! => {"changed": false, "msg": "Unable get authentication flow browser-custom: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)>"}
```

after:
```paste below
TASK [keycloak : create custom authentication flow] *******************************************************************************************************************************************************************************************
ok: [[********]:]
```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! It looks like these got forgotten. Can you please add [a changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: changelogs/fragments/4382-keycloak-add-missing-validate_certs-parameters.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+bugfixes:
+  - keycloak_* - the documented `validate_certs` parameter was not taken into
+    account when calling the `open_url` function, thus enforcing certificate
+    validation even when `validate_certs` was set to `no`.
```
Comment (felixfontein) [inline changelogs/fragments/4382-keycloak-add-missing-validate_certs-parameters.yml:5]:
```suggestion
  - keycloak_* - the documented ``validate_certs`` parameter was not taken into
    account when calling the ``open_url`` function in some cases, thus enforcing
    certificate validation even when ``validate_certs`` was set to ``false``.
```

Comment (lmeunier) [reply changelogs/fragments/4382-keycloak-add-missing-validate_certs-parameters.yml]: Thanks for the suggestion. It just has been applied.

**Review by felixfontein — APPROVED** [review]
Looks good to me. If nobody complains, I'll merge beginning of next week.

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@lmeunier thanks for fixing this!
@russoz thanks for reviewing!

**Pull request merged**